<launch>
    <param name="use_sim_time" value="true"/>
    <!--注意使用 仿真时间  bag播放时加-clock-->
    <!--default定义的值可以在launch的时候通过命令行指定而被覆盖,而value定义的值不会-->
    <!--    <arg name="ndt_resolution" default="3.0"/>
        <arg name="ndt_step_size" default="0.1"/>
        <arg name="ndt_trans_eps" default="0.01"/>
        <arg name="ndt_max_iter" default="30"/>
        <arg name="voxel_leaf_size" default="0.1"/>
        <arg name="min_scan_range" default="0.5"/>
        <arg name="max_scan_range" default="80.0"/>
        <arg name="min_add_scan_shift" default="0.5"/>
        <arg name="max_submap_size" default="8"/>-->

    <arg name="init_x" default="0"/>
    <arg name="init_y" default="0"/>
    <arg name="init_z" default="0"/>
    <arg name="init_roll" default="0"/>
    <arg name="init_pitch" default="0"/>
    <arg name="init_yaw" default="0"/>

    <!--    <include file="$(find vehicle_description)/launch/lexus.launch"/>-->
    <!--    <node pkg="tf" type="static_transform_publisher" name="word2map_tf" args="0 0 0 0 0 0 /world /map 10"/>-->

    <!--    <node pkg="tf" type="static_transform_publisher" name="rslidar" args="0 0 0 0 0 0 /base_link /velo_link 100"/>-->

    <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find xchu_mapping)/rviz/mapping.rviz"/>

    <node pkg="xchu_mapping" type="filter_node" name="filter_node" output="screen"/>

    <node pkg="xchu_mapping" type="xchu_odom_node" name="xchu_odom_node" output="screen">
        <param name="map_saved_dir" value="$(find xchu_mapping)/pcd/"/>
        <!--<param name="method_type"  value="1"  type="int"/>  -->
        <param name="use_imu" value="false" type="bool"/>
        <param name="use_odom" value="false" type="bool"/>
        <param name="imu_upside_down" value="false" type="bool"/>
        <!--      /imu/data  /kitti/oxts/imu /kitti/oxts/gps/fix -->
        <param name="imu_topic" value="/kitti/oxts/imu" type="str"/>
        <param name="odom_topic" value="/golfcar/odom" type="str"/>

        <param name="ndt_method_type" value="3" type="int"/> <!--pcl:0,cpu:1,gpu:2,omp:3,gpu_ptr:4-->
        <param name="incremental_voxel_update" value="true" type="bool"/>
    </node>

    <!--    <node pkg="xchu_mapping" type="mapping_node" name="mapping_node" output="screen"/>\-->

    <node pkg="xchu_mapping" type="loop_node" name="loop_node" output="screen"/>

    <node pkg="xchu_mapping" type="optimize_gtsam_node" name="optimize_gtsam_node" output="screen"/>

    <!--    <node pkg="xchu_mapping" type="optimize_node" name="optimize_node" output="screen"/>-->


</launch>
